### PR TITLE
Do not panic when doing f/resize

### PR DESCRIPTION
**Pull request by BourgondAries**
How to reproduce: Move the cursor to the top-right quadrant of the 128
128 image. Run `:f/resize 1 1`, run `:f/resize 128 128`

This would cause an underflow in the `color_at` function.

The test case does not reproduce the bug so I'd like some help with
that. Overall this library needs more testing.

### CODE REVIEW

File: src/session.rs [hunk 1]
```diff
@@ -2152,23 +2152,31 @@ impl Session {
...
-        Rgba8::new(bgra.r, bgra.g, bgra.b, bgra.a)
+        if let Some(index) = index {
+            if let Some(bgra) = pixels.get(index) {
+                Some(Rgba8::new(bgra.r, bgra.g, bgra.b, bgra.a))
+            } else {
+                None
+            }
+        } else {
+            None
+        }
```
Comment (cloudhead) [inline src/session.rs]: Can we perhaps make this a little tighter by using `and_then` to get rid of the `else` branches?

File: src/session.rs [hunk 2]
```diff
@@ -2152,23 +2152,31 @@ impl Session {
     }
 
     /// Get the color at the given view coordinate.
-    fn color_at(&self, v: ViewId, p: ViewCoords<u32>) -> Rgba8 {
+    fn color_at(&self, v: ViewId, p: ViewCoords<u32>) -> Option<Rgba8> {
         let resources = self.resources.lock();
         let snapshot = resources.get_snapshot(&v);
 
-        assert!(snapshot.len() % std::mem::size_of::<Bgra8>() == 0);
```
Comment (cloudhead) [inline src/session.rs]: Any reason why this was removed?

Comment (BourgondAries) [reply src/session.rs]: Because a snapshot being 1x1 has a len of 1, so this assertion makes no sense.

Comment (cloudhead) [reply src/session.rs]: Yeah - you're right. This is actually a regression - `len()` is supposed to return the number of bytes, which would mean it is always at least `4`.

File: src/session.rs (hunk 1 above)
Comment (BourgondAries) [reply src/session.rs]: Sure!

**Comment by cloudhead** [comment]
As for the reason the tests are passing - it's because the renderer creates the snapshots from the framebuffer - so in that test this isn't happening and thus the snapshot is always the initial 128x128 one. I'd suggest removing the tests for now, as there will be a much more comprehensive way of testing things soon.

**Comment by BourgondAries** [comment]
@cloudhead alright, test removed.

**Comment by cloudhead** [comment]
Thanks!

**Pull request merged**